src/compiler/factory/nodeChildren.ts

```diff
@@ -50,3 +50,12 @@ export function unsetNodeChildren(node: Node, origSourceFile: SourceFileLike) {
     }
     sourceFileToNodeChildren.get(origSourceFile)?.delete(node);
 }
+
+/** @internal */
+export function transferSourceFileChildren(sourceFile: SourceFileLike, targetSourceFile: SourceFileLike) {
+    const map = sourceFileToNodeChildren.get(sourceFile);
+    if (map !== undefined) {
+        sourceFileToNodeChildren.delete(sourceFile);

```

Comment:
Is it harmful to leave the old entry behind? Theoretically it will get GC'd once the old one goes away, but I guess maybe it's bad to have two people able to write to this? (Theoretically their keys would be distinct anyway?)

Reply:
That's not a bad reason, but I guess the main thing I was thinking about is avoiding a `set` resizing the table if possible. It's probably a small cost, and not bullet-proof, but there's no reason for the two `SourceFile`s to take up space in the map if they're not being used.

Technically you could argue we never need to call `unsetNodeChildren` either since the node will never be reused.

Reply:
Since WeakMap is not iterable and therefore doesn't have a defined order of keys, I'm unclear on whether or not deleting entries from a WeakMap is a trivial operation. It's an interesting question which way performance goes on trimming a WeakMap vs. letting the values get GC'd, but there is something to be said for cleanliness.